handler/src/main/java/io/netty/handler/ipfilter/IpSubnetFilter.java

```diff
@@ -160,13 +160,13 @@ public IpSubnetFilter(boolean acceptIfNotFound, List<IpSubnetFilterRule> rules)
             ipFilterRuleTypeIPv6 = null;
         }
 
-        this.ipv4Rules = sortAndFilter(unsortedIPv4Rules);
-        this.ipv6Rules = sortAndFilter(unsortedIPv6Rules);
+        this.ipv4Rules = unsortedIPv4Rules.isEmpty() ? null : sortAndFilter(unsortedIPv4Rules);
+        this.ipv6Rules = unsortedIPv6Rules.isEmpty() ? null : sortAndFilter(unsortedIPv6Rules);
     }
 
     @Override
     protected boolean accept(ChannelHandlerContext ctx, InetSocketAddress remoteAddress) {
-        if (remoteAddress.getAddress() instanceof Inet4Address) {
+        if (ipv4Rules != null && remoteAddress.getAddress() instanceof Inet4Address) {
```

Comment:
It also makes sense to use an array instead of List for `ipv4Rules` and `ipv6Rules` -> less memory usage, and better performance.

Reply:
Definitely!